src/index.ts

```diff
@@ -24,20 +24,29 @@ export const createApi = (gitlabToken?: string) => {
   }
 
   const token = gitlabToken || process.env.GITLAB_TOKEN
+  if (!token) {
+    throw new Error('GitLab token is not set')
+  }
 
-  let tokenType: 'jobToken' | 'oauthToken' | 'token' = 'token'
+  const host = process.env.GITLAB_HOST ?? process.env.CI_SERVER_URL
 
+  // we cannot use { [tokenType]: token } now
```

Comment:
It weird, can you show some related typing errors here?

Reply:
![image](https://github.com/un-ts/changesets-gitlab/assets/70696274/269d7427-fa85-4a4a-b572-0f49cdbaa030)

```
Argument of type '{ [x: string]: string | undefined; host: string | undefined; }' is not assignable to parameter of type 'BaseResourceOptions<false>'.
  Property 'jobToken' is missing in type '{ [x: string]: string | undefined; host: string | undefined; }' but required in type 'BaseRequestOptionsWithJobToken<false>'.
```

Reply:
OK, weird but acceptable.